custom/enikshay/tasks.py

```diff
@@ -0,0 +1,180 @@
+            update = EpisodeUpdate(episode, self)
+            case_block = update.case_block()
+            if case_block:
+                submit_case_blocks(
+                    [ElementTree.tostring(case_block.as_xml())],
+                    self.domain
+                )
+
+    def _get_open_episode_cases(self):
+        case_accessor = CaseAccessors(self.domain)
+        case_ids = case_accessor.get_open_case_ids_in_domain_by_type(CASE_TYPE_EPISODE)
+        return case_accessor.iter_cases(case_ids)
+
+    @memoized
+    def get_doses_data(self):
+        # return 'doses_per_week' by 'schedule_id' from the Fixture data
+        fixtures = FixtureDataItem.get_indexed_items(self.domain, DAILY_SCHEDULE_FIXTURE_NAME, SCHEDULE_ID_FIXTURE)
+        return dict((k, int(fixture['doses_per_week'])) for k, fixture in fixtures.items())
+
+
+def index_by_adherence_date(adherence_cases):
+    """
+    index by day of adherence_date datetime
+    """
+    by_date = defaultdict(list)
+    for case in adherence_cases:
+        adherence_date = parse_datetime(case.dynamic_case_properties().get('adherence_date')).date()
+        by_date[adherence_date].append(case)
+    return by_date
+
+
+class EpisodeUpdate(object):
+
+    def __init__(self, episode_case, case_updater):
+        self.episode = episode_case
+        self.case_updater = case_updater
+
+    def get_property(self, property):
+        return self.episode.dynamic_case_properties().get(property)
+
+    def get_latest_adherence_case_for_episode(self):
+        """
+        return open case of type 'adherence' reversed-indexed to episode and
+            with latest 'adherence_date' property
+        """
+        case_accessor = CaseAccessors(self.case_updater.domain)
+        indexed_cases = case_accessor.get_reverse_indexed_cases([self.episode.case_id])
+        # sometime far back
+        latest_date = pytz.UTC.localize(datetime.datetime(1990, 1, 1))
+        latest_case = None
+        for case in indexed_cases:
+            adherence_date = parse_datetime(case.dynamic_case_properties().get('adherence_date'))
+            if (not case.closed and
+               case.type == CASE_TYPE_ADHERENCE and
+               adherence_date > latest_date):
+                latest_date = adherence_date
+                latest_case = case
+        return latest_case
+
+    def update_json(self):
+        adherence_schedule_date_start = parse_datetime(self.get_property('adherence_schedule_date_start'))
+        if not adherence_schedule_date_start:
+            # adherence schedule hasn't been selected, so no update necessary
+            return {}
+
+        if adherence_schedule_date_start > self.case_updater.purge_date:
+            return {
+                'aggregated_score_date_calculated': adherence_schedule_date_start - datetime.timedelta(days=1),
+                'expected_doses_taken': 0,
+                'aggregated_score_count_taken': 0
+            }
```

Comment:
(although I think i realized later that these are case properties that get passed to `update` so maybe not worth it)